python/src/rtree.rs

```diff
@@ -27,6 +29,228 @@ impl<'a> FromPyObject<'a> for RTreeMethod {
     }
 }
 
+struct PyU8Buffer(PyBuffer<u8>);
+
+impl<'py> FromPyObject<'py> for PyU8Buffer {
+    fn extract_bound(obj: &Bound<'py, PyAny>) -> PyResult<Self> {
+        let buffer = PyBuffer::<u8>::get_bound(obj)?;
+        if !buffer.readonly() {
+            return Err(PyValueError::new_err("Must be read-only byte buffer."));
+        }
+        if buffer.dimensions() != 1 {
+            return Err(PyValueError::new_err("Expected 1-dimensional array."));
+        }
+        // Note: this is probably superfluous for 1D array
+        if !buffer.is_c_contiguous() {
+            return Err(PyValueError::new_err("Expected c-contiguous array."));
+        }
+        if buffer.len_bytes() == 0 {
+            return Err(PyValueError::new_err("Buffer has no data."));
+        }
+
+        Ok(Self(buffer))
+    }
+}
+
+impl AsRef<[u8]> for PyU8Buffer {
+    fn as_ref(&self) -> &[u8] {
+        let len = self.0.item_count();
+        let data = self.0.buf_ptr() as *const u8;
+        unsafe { std::slice::from_raw_parts(data, len) }
```

Comment:
@H-Plus-Time this seems much cleaner than doing a transmute, because I believe the transmute relies on `#[repr(transparent)]`